Applications/Spire/Source/Ui/DropDownList.cpp

```diff
@@ -0,0 +1,255 @@
+  m_layout = new QVBoxLayout(main_widget);
+  m_layout->setContentsMargins({});
+  m_layout->setSpacing(0);
+  m_scroll_area->setWidget(main_widget);
+  initialize_widget(m_scroll_area);
+  set_items(items);
+  parent->installEventFilter(this);
+}
+
+
+
+bool DropDownList::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == parent()) {
+    if(event->type() == QEvent::KeyPress) {
+      auto e = static_cast<QKeyEvent*>(event);
+      switch(e->key()) {
+        case Qt::Key_Down:
+          if(isVisible()) {
+            activate_next();
+            return true;
+          }
+          break;
+        case Qt::Key_Up:
+          if(isVisible()) {
+            activate_previous();
+            return true;
+          }
+          break;
+        case Qt::Key_Enter:
+        case Qt::Key_Return:
+          if(isVisible() && m_highlight_index) {
+            on_item_selected(get_widget(*m_highlight_index)->get_value(),
+              *m_highlight_index);
+            return true;
+          }
+          break;
+      }
+    }
+  }
+  return DropDownWindow::eventFilter(watched, event);
+}
+
+void DropDownList::hideEvent(QHideEvent* event) {
+  m_scroll_area->verticalScrollBar()->setValue(0);
+  if(m_highlight_index) {
+    auto item = m_layout->itemAt(*m_highlight_index);
+    if(item != nullptr) {
+      static_cast<DropDownItem*>(item->widget())->reset_highlight();
+    }
+    m_highlight_index = boost::none;
+  }
+}
+
+void DropDownList::keyPressEvent(QKeyEvent* event) {
+  switch(event->key()) {
+    case Qt::Key_Down:
+      if(isVisible()) {
+        activate_next();
+      }
+      break;
+    case Qt::Key_Enter:
+      if(m_highlight_index) {
+        on_item_selected(get_widget(*m_highlight_index)->get_value(),
+          *m_highlight_index);
+      }
+      break;
+    case Qt::Key_Escape:
+      hide();
+      break;
+    case Qt::Key_Up:
+      if(isVisible()) {
+        activate_previous();
+      }
+      break;
+  }
+  DropDownWindow::keyPressEvent(event);
+}
+
+connection DropDownList::connect_activated_signal(
+    const ActivatedSignal::slot_type& slot) const {
+  return m_activated_signal.connect(slot);
+}
+
+connection DropDownList::connect_highlighted_signal(
+    const HighlightedSignal::slot_type& slot) const {
+  return m_highlighted_signal.connect(slot);
+}
+
+connection DropDownList::connect_index_selected_signal(
+    const IndexSelectedSignal::slot_type& slot) const {
+  return m_index_selected_signal.connect(slot);
+}
+
+connection DropDownList::connect_value_selected_signal(
+    const ValueSelectedSignal::slot_type& slot) const {
+  return m_value_selected_signal.connect(slot);
+}
+
+QVariant DropDownList::get_value(int index) {
+  if(m_layout->count() - 1 < index) {
+    return QVariant();
+  }
+  return get_widget(index)->get_value();
+}
+
+void DropDownList::insert_item(DropDownItem* item) {
+  m_item_selected_connections.push_back(item->connect_selected_signal(
+    [=] (auto value) {
```

Comment:
You will likely end up taking an ```auto& value```, and then removing the ```std::move``` on line 132. This will happen when the signal is changed to pass a ```const &``` instead of passing by value.